compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/TransactionImportValidator.java

```diff
@@ -79,6 +80,10 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
         super.visit(functionDefinitionNode);
     }
 
+    @Override
+    public void visit(TransactionalExpressionNode transactionalExpressionNode) {
+        importTransactionPackage = true;
```

Comment:
the only test scenario that was not covered was, having transactional expr in the code without having the transaction block.